js/src/index.js

```diff
@@ -13,5 +13,12 @@ require('leaflet/dist/images/marker-icon.png');
 require('leaflet/dist/images/marker-icon-2x.png');
 
 // Export everything from jupyter-leaflet and the npm package version number.
+hasL = (typeof(window.L) != 'undefined');
 module.exports = require('./jupyter-leaflet.js');
 module.exports['version'] = require('../package.json').version;
+
+if (hasL) {
+    console.log("Existing `L` detected, running ipyleaflet's Leaflet in no-conflict mode as `ipyL`");
+    ipyL = L.noConflict();
```

Comment:
I'm not entirely sure I understand. Shouldn't we use `ipyL` everywhere in ipyleaflet in this case? Instead of `L`? I mean, what do we do with this variable?

Reply:
The ipyleaflet code will still use L since when it is all loaded L is the internal Leaflet namespace within the module.

Leaflet's no-conflict storage of the existing global L is independent of any module exports. So that's why we first check if there will be a conflict, then load ipyleaflet, then restore the original global L.

I have only tested using the ipyleaflet functions we use. Without a test suite it might be better to pull this change initially to its own branch for others to check.

Reply:
So, if I understand correctly, if you want to restore the global `L`, shouldn't this line of code be: `L = L.noConflict();` or `window.L = L.noConflict();`?

Reply:
L. noConflict() restores the previous global L to `window.L`, and returns the current Leaflet library so you can assign it to a different variable name:

https://leafletjs.com/reference-1.4.0.html#noconflict

Reply:
Ok perfect then :) Thanks!

Reply:
Thanks Martin!